### ISSUE TITLE
Beacon node segfaults on Sepolia and Prater

### ISSUE THREAD

**Issue by jakubgs** [status-im/nim-beacon-chain#5693]
**Describe the bug**
Current we have two nodes stuck in a segfault loop on two hosts:

* `rocketpool-eth2-node`@`metal-01.ih-eu-mda1.nimbus.eth1`
    - Running `v24.1.0-47e035` and possibly other versions but unconfirmed.
* `beacon-node-sepolia-unstable-trial-01`@`linux-01.ih-eu-mda1.nimbus.sepolia`
    - Running `v23.11.0-7db95f` as well as `v24.1.0-3f525a`.

**Logs**
```
admin@linux-01.ih-eu-mda1.nimbus.sepolia:~ % sudo dmesg | grep -C1 segfault | tail -n6
[2413522.190505] nimbus_beacon_n[1327034]: segfault at 0 ip 000055d2be06ef8b sp 00007ffef5917590 error 4 in nimbus_beacon_node_3f525acb[55d2bd86e000+1147000]
[2413522.190515] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f b0 a0 01 00 00 c5 f9 7f 36 c5 f9 6f b8 b0 01 00 00 c5 f9 7f 7e 10 48 8b 83 c0 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 c0 01 00 00 48 8b 80 18 02 00 00 48
[2413526.653988] nimbus_beacon_n[1327063]: segfault at 0 ip 000055ee4aba5f8b sp 00007ffed9b20670 error 4 in nimbus_beacon_node_3f525acb[55ee4a3a5000+1147000]
[2413526.654026] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f b0 a0 01 00 00 c5 f9 7f 36 c5 f9 6f b8 b0 01 00 00 c5 f9 7f 7e 10 48 8b 83 c0 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 c0 01 00 00 48 8b 80 18 02 00 00 48
[2413531.312088] nimbus_beacon_n[1327095]: segfault at 0 ip 0000561a5a469f8b sp 00007ffcf76a7a10 error 4 in nimbus_beacon_node_3f525acb[561a59c69000+1147000]
[2413531.312100] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f b0 a0 01 00 00 c5 f9 7f 36 c5 f9 6f b8 b0 01 00 00 c5 f9 7f 7e 10 48 8b 83 c0 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 c0 01 00 00 48 8b 80 18 02 00 00 4
```
```
admin@metal-01.ih-eu-mda1.nimbus.eth1:~ % sudo dmesg -T | grep -C1 segfault | tail -n6
[Fri Jan  5 14:44:52 2024] nimbus_beacon_n[2134913]: segfault at 0 ip 00005641c82bfe83 sp 00007fffae811330 error 4 in nimbus_beacon_node_47e035f3[5641c7aaf000+10a2000]
[Fri Jan  5 14:44:52 2024] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f 98 a0 01 00 00 c5 f9 7f 1e c5 f9 6f a0 b0 01 00 00 c5 f9 7f 66 10 48 8b 83 b8 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 b8 01 00 00 48 8b 80 18 02 00 00 48
[Fri Jan  5 14:45:05 2024] nimbus_beacon_n[2134931]: segfault at 0 ip 0000559841180e83 sp 00007ffcc22ff090 error 4 in nimbus_beacon_node_47e035f3[559840970000+10a2000]
[Fri Jan  5 14:45:05 2024] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f 98 a0 01 00 00 c5 f9 7f 1e c5 f9 6f a0 b0 01 00 00 c5 f9 7f 66 10 48 8b 83 b8 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 b8 01 00 00 48 8b 80 18 02 00 00 48
[Fri Jan  5 14:45:17 2024] nimbus_beacon_n[2134944]: segfault at 0 ip 0000555e42754e83 sp 00007ffce84e5150 error 4 in nimbus_beacon_node_47e035f3[555e41f44000+10a2000]
[Fri Jan  5 14:45:17 2024] Code: 89 c6 48 8b 43 58 48 8b 40 18 c5 f9 6f 98 a0 01 00 00 c5 f9 7f 1e c5 f9 6f a0 b0 01 00 00 c5 f9 7f 66 10 48 8b 83 b8 01 00 00 <48> 8b 00 48 89 46 20 48 8b 83 b8 01 00 00 48 8b 80 18 02 00 00 48
```

**Additional context**
The issues on `nimbus.eth1` started yesterday on `Jan 04 20:07:58`:
```
admin@metal-01.ih-eu-mda1.nimbus.eth1:~ % j --since '30 days ago' --grep segfault | head -n1
Jan 04 20:07:58 metal-01.ih-eu-mda1.nimbus.eth1 kernel: nimbus_beacon_n[1968747]: segfault at 0 ip 000055f09ee5ee83 sp 00007ffcf36260d0 error 4 in nimbus_beacon_node_47e035f3[55f09e64e000+10a2000]
```
And the ones on `nimbus.sepolia` on `Dec 22 22:13:12`:
```
admin@linux-01.ih-eu-mda1.nimbus.sepolia:~ % j --since '30 days ago' --grep segfault | head -n1
Dec 22 22:13:12 linux-01.ih-eu-mda1.nimbus.sepolia kernel: nimbus_beacon_n[3026096]: segfault at 0 ip 0000564b977ae083 sp 00007fffe8fee730 error 4 in nimbus_beacon_node_6f32e89f[564b96f9d000+10a1000]
```
But that cannot be trusted as that's actually the limit of logs.

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth2/pull/5695 and released in v24.1.1.

**Comment by jakubgs** [comment]
Thanks for the quick fix.
